qiskit/circuit/quantumcircuit.py

```diff
@@ -4077,6 +4087,21 @@ def pauli(
 
         return self.append(PauliGate(pauli_string), qubits, [])
 
+    def permutation(self, pattern: List[int], qubits: Sequence[QubitSpecifier]) -> InstructionSet:
```

Comment:
Do we really need a circuit method for the permutation like this? We typically only do this for standard gates, is there a pressing need to do this? If not I'd rather defer this for now, if the need arises later it's easy to add. But, it's hard to roll this back if we decided it's not really useful in the future.

Reply:
There is definitely no pressing need for this, I thought that the convention was to add a method to QuantumCircuit for every object that inherits from Gate; sure, I can remove it.

Reply:
Done in [e1e9f05](https://github.com/Qiskit/qiskit-terra/pull/9157/commits/e1e9f05de3615f2fc9b2ec55696061304b9d4ef8)